src/main/java/org/assertj/core/api/AbstractPathAssert.java

```diff
@@ -0,0 +1,107 @@
+package org.assertj.core.api;
+
+import org.assertj.core.internal.Paths;
+import org.assertj.core.util.VisibleForTesting;
+
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+
+/**
+ * Assertions for {@link Path} objects
+ *
+ * <p>Please note that a {@code Path} object may, or may not, be issued from the
+ * {@link FileSystems#getDefault() default filesystem} (that is, the filesystem
+ * which the JVM is running on). These assertions are "filesystem independent".
+ * </p>
+ *
+ * @param <S> self type
+ *
+ * @see Path
+ * @see FileSystem
+ * @see FileSystem#getPath(String, String...)
+ * @see Files
+ */
+public abstract class AbstractPathAssert<S extends AbstractPathAssert<S>>
+    extends AbstractAssert<S, Path>
+{
+    @VisibleForTesting
+    protected Paths paths = Paths.instance();
+
+    protected AbstractPathAssert(final Path actual, final Class<?> selfType)
+    {
+        super(actual, selfType);
+    }
+
+    /**
+     * Assert that a given path exists
+     *
+     * <p><strong>IMPORTANT NOTE:</strong> this method will NOT follow symbolic
+     * links (provided that the underlying {@link FileSystem} of this path
+     * supports symbolic links at all).</p>
+     *
+     * <p>This means that even if the path in question is a symbolic link
+     * but its target is nonexistent, this assertion will
+     * <strong>succeed</strong>.</p>
+     *
```

Comment:
add code example including  standard file and a symbolic link to illustrate the comment